processing/src/main/java/io/druid/query/aggregation/DoubleMaxAggregatorFactory.java

```diff
@@ -157,7 +161,9 @@ public String getName()
   @Override
   public List<String> requiredFields()
   {
-    return fieldName != null ? Collections.singletonList(fieldName) : Parser.findRequiredBindings(expression);
+    return fieldName != null
+           ? Collections.singletonList(fieldName)
+           : Parser.findRequiredBindings(Parser.parse(expression, macroTable));
```

Comment:
It looks to be possible to keep parse result if ```fieldName``` is null in the constructor. Same for other ```AggregatorFactory```s.